### PR TITLE
⚠️ Use objects generated by KCP, MachineSets and MachinePools will now consistently use machine name

### LINKED ISSUE #9733: Generate Machine / InfraMachine / BootstrapConfig with the same name
Today we generate separate random names for Machine / InfraMachine / BootstrapConfig in KCP and the MachineSet controller. This makes it unnecessarily complicated to correlate these objects as an operator.

The idea is that we should only create one random name on Machine create and then use it for all 3 objects

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
(Compared to the issue) I also implemented it for MachinePool Machines. There is nothing we can do about BootstrapConfig there because it belongs to the MachinePool not to individual Machines, but at least Machines and InfraMachines now have the same name.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #9733

### CODE REVIEW

**Comment by chrischdi** [comment]
Small improvement nit.

As of that
/lgtm

File: controllers/external/util.go [hunk 1]
```diff
@@ -157,6 +166,9 @@ func GenerateTemplate(in *GenerateTemplateInput) (*unstructured.Unstructured, er
 	to.SetUID("")
 	to.SetSelfLink("")
 	to.SetName(names.SimpleNameGenerator.GenerateName(in.Template.GetName() + "-"))
+	if in.Name != "" {
+		to.SetName(in.Name)
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply controllers/external/util.go]: Nice idea. Done!

**Comment by sbueringer** [comment]
/assign @killianmuldoon 

If you have some time

**Comment by sbueringer** [comment]
Rebased

/assign @chrischdi @fabriziopandini

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
/lgtm

Thanks! Name consistency will greatly improve triaging and investigating issues, as well as navigating the CAPI object hierarchy

**Pull request merged**